### PR TITLE
Metric Servers management in Proxmox

### LINKED ISSUE #259: Feature Request: Add Support for Managing Proxmox Metric Servers
I would like to request the addition of support for managing Proxmox Metric Servers within this role.

One possible approach to implement this feature would be to create a template for the /etc/pve/status.cfg file and restart the pvestatd service. However, this solution does not allow for the management of passwords credentials.

An alternative and more secure method would be to add support using the Proxmox API, which would allow for full management, including password handling. Unfortunately, I am less familiar with this approach, so I'm unable to propose the code myself.

Thank you for considering this feature!

### PR DESCRIPTION
**Pull request by thystips**
Tested only with PVE 7

Telegraf can be used to transfert metrics to Prometheus, Elasticsearch, etc.

Resolve #259

### CODE REVIEW

File: library/proxmox_metric_server.py [hunk 1]
```diff
@@ -0,0 +1,411 @@
...
+        self.path = module.params["path"]
+        self.api_path_prefix = module.params["api_path_prefix"]
+        self.timeout = module.params["timeout"]
+        self.max_body_size = module.params["max_body_size"]
+        self.mtu = module.params["mtu"]
+        self.verify_certificate = module.params["verify_certificate"]
+        self.state = module.params["state"]
+
+        try:
+            self.existing_server = pvesh.get("cluster/metrics/server")
```
Comment (lae) [inline library/proxmox_metric_server.py:149]: minor nitpick, but pluralize var?

File: library/proxmox_metric_server.py [hunk 2]
```diff
@@ -0,0 +1,411 @@
...
+
+        self.parse_servers()
+
+    def parse_servers(self):
+        self.servers = []
+        for existing_server in self.existing_server: # type: ignore
+            self.servers.append(existing_server.get("id"))
+
+    def lookup(self):
+        self.servers = []
```
Comment (lae) [inline library/proxmox_metric_server.py:161]: unused variable

**Review by lae — CHANGES_REQUESTED** [review]
Overall very well done. Thanks for the PR!

There's one critical change needed (the imports) and I've left comments for some other things.

**Comment by lae** [comment]
oh, feel free to add your name to the list of contributors for metric server configuration support. (otherwise I'll add it later)

**Comment by thystips** [comment]
I initially write this module for a private collection where this module is used as standalone after proxmox install, I forgot edit some code like import. It' s fixed.
All requests solved.

File: library/proxmox_metric_server.py [hunk 3]
```diff
@@ -340,10 +338,9 @@ def main():
             msg="The 'protocol' parameter must be 'tcp' or 'udp' for 'graphite' type."
         )
 
-    if module.params["type"] == "graphite" and (
-        module.params["bucket"] is not None
-        or module.params["organization"] is not None,
-        module.params["token"] is not None,
+    if module.params["type"] == "graphite" and any(
```
Comment (lae) [inline library/proxmox_metric_server.py:341]: I'm assuming it's intentional, but the conditions in this statement slightly changed ( `a || b && c` → `a || b || c` ).

**Review by lae — APPROVED** [review]

File: library/proxmox_metric_server.py (hunk 3 above)
Comment (thystips) [reply library/proxmox_metric_server.py:341]: Yes it is intentional. Graphite have less arguments in API.

**Pull request merged**